debouncer_test.go

```diff
@@ -216,17 +216,60 @@ func TestDone(t *testing.T) {
 	debouncer := godebouncer.New(200 * time.Millisecond).WithTriggered(incrementCount)
 	expectedCounter := int(2)
 
-	fmt.Println("Action 1")
 	debouncer.SendSignal()
-	time.Sleep(400 * time.Millisecond)
+	<-debouncer.Done()
 
-	fmt.Println("Action 2")
 	debouncer.SendSignal()
 	<-debouncer.Done()
 
-	fmt.Println(len(debouncer.Done()))
+	if *countPtr != expectedCounter {
+		t.Errorf("Expected count %d, was %d", expectedCounter, *countPtr)
+	}
+}
+
+func TestDoneInGoroutine(t *testing.T) {
+	countPtr, incrementCount := createIncrementCount(0)
+	debouncer := godebouncer.New(200 * time.Millisecond).WithTriggered(incrementCount)
+	expectedCounter := int(3)
+
+	debouncer.SendSignal()
+	go func() {
+		<-debouncer.Done() // awaits for the second send signal to complete
+		*countPtr += 2
+	}()
+
+	debouncer.SendSignal() // after 200 milliseconds, unblock done channel in 2 goroutines
+	<-debouncer.Done()
+
+	time.Sleep(100 * time.Millisecond)
 
 	if *countPtr != expectedCounter {
 		t.Errorf("Expected count %d, was %d", expectedCounter, *countPtr)
 	}
```

Comment:
Add 1 more case about call Done twice times.

SendSignal()
<- Done()

SendSignal()
<- Done()

it should be unblock 2 times to make sure 2 sequence of done should work normally.

Reply:
There is already a test case for calling Done() twice:
![CleanShot 2022-05-19 at 12 14 57](https://user-images.githubusercontent.com/17645203/169214412-946e6357-314c-459a-8a0b-264ed18d20d7.png)

Reply:
ah, it's ok, I missed it.

do we have case with following steps:
```
SendSignal()
<- Done()
<- Done() // it should be hang 
```

Reply:
I've add 2 cases:
```go
// TestDoneHangBeforeSendSignal
d := godebouncer.New(wait).WithTriggered(func)
<-d.Done() // should hang
```
and
```go
// TestDoneHangIfBeingCalledTwice
d := godebouncer.New(wait).WithTriggered(func)
d.SendSignal()
<-d.Done()
<-d.Done() // should hang
```